pkg/operator/configobservation/images/observe_images.go

```diff
@@ -148,6 +148,46 @@ func ObserveAllowedRegistriesForImport(genericListers configobserver.Listers, re
 	return observedConfig, errs
 }
 
+func ObserveImagestreamImportMode(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	imageStreamImportModePath := []string{"imagePolicyConfig", "imageStreamImportMode"}
+	currentImageStreamImportMode, _, err := unstructured.NestedString(existingConfig, imageStreamImportModePath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentImageStreamImportMode) > 0 {
+		err := unstructured.SetNestedField(prevObservedConfig, currentImageStreamImportMode, imageStreamImportModePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	// now gather the cluster config and turn it into the observed config
+	observedConfig := map[string]interface{}{}
+	configImage, err := listers.ImageConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("image.config.openshift.io/cluster: not found")
```

Comment:
can we include more contextual information here, like the observer name?

Reply:
added.